processor/src/lib.rs

```diff
@@ -128,7 +128,7 @@ pub struct Process {
     stack: Stack,
     range: RangeChecker,
     chiplets: Chiplets,
-    advice: AdviceProvider,
+    advice: BaseAdviceProvider,
 }
```

Comment:
I was thinking we'd make `Process` generic in this PR. Basically, we still keep signatures of `execute()` and `execute_iter()` functions as they are now, but `Process` would be updated to be generic. Then, in these functions we'd instantiate `BaseAdviceProvider` and pass it to `Process::new()` and `Process::new_debug()` constructors (and for now, we can just clone `inputs`).

For example, signature of `Process::new()` could look like:
```Rust
pub fn new(kernel: &Kernel, inputs: ProgramInputs, advice: A) -> Self
```
And then in subsequent PRs we'll update the type of `inputs` to `StackInputs` or something like that.

Reply:
Done! There is a small change in `execute_iter` as `VmStateIterator` needs to be aware of the concrete implementation for its process.

I opted to leave it as fixed to `MemAdviceProvider` there so we keep it simple for now, and this can be changes to generic when we update the public interfaces.